ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/PolylineDemoActivity.kt

```diff
@@ -0,0 +1,291 @@
+            max = MAX_WIDTH_PX
+            progress = MAX_WIDTH_PX / 2
+        }
+
+        startCapSpinner = findViewById<Spinner>(R.id.startCapSpinner).apply {
+            adapter = ArrayAdapter(
+                    this@PolylineDemoActivity, android.R.layout.simple_spinner_item,
+                    getResourceStrings(capTypeNameResourceIds))
+        }
+
+        endCapSpinner = findViewById<Spinner>(R.id.endCapSpinner).apply {
+            adapter = ArrayAdapter(
+                    this@PolylineDemoActivity, android.R.layout.simple_spinner_item,
+                    getResourceStrings(capTypeNameResourceIds))
+        }
+
+        jointTypeSpinner = findViewById<Spinner>(R.id.jointTypeSpinner).apply {
+            adapter = ArrayAdapter(
+                    this@PolylineDemoActivity, android.R.layout.simple_spinner_item,
+                    getResourceStrings(jointTypeNameResourceIds))
+        }
+
+        patternSpinner = findViewById<Spinner>(R.id.patternSpinner).apply {
+            adapter = ArrayAdapter(
+                    this@PolylineDemoActivity, android.R.layout.simple_spinner_item,
+                    getResourceStrings(patternTypeNameResourceIds))
+        }
+
+        clickabilityCheckbox = findViewById<CheckBox>(R.id.toggleClickability)
+
+        val mapFragment = supportFragmentManager.findFragmentById(R.id.map) as SupportMapFragment
+        mapFragment.getMapAsync(this)
+    }
+
+    private fun getResourceStrings(resourceIds: IntArray): List<String> {
+        return resourceIds.map { getString(it) }
+    }
+
+    override fun onMapReady(googleMap: GoogleMap?) {
+
+        // exit early if the map was not initialised properly
+        googleMap ?: return
+
+        with(googleMap) {
+            // Override the default content description on the view, for accessibility mode.
+            setContentDescription(getString(R.string.polyline_demo_description))
+
+            // A geodesic polyline that goes around the world.
+            addPolyline(PolylineOptions().apply {
+                add(lhrLatLng, aklLatLng, laxLatLng, jfkLatLng, lhrLatLng)
+                width(INITIAL_STROKE_WIDTH_PX.toFloat())
+                color(Color.BLUE)
+                geodesic(true)
+                clickable(clickabilityCheckbox.isChecked)
+            })
+
+            // Move the googleMap so that it is centered on the mutable polyline.
+            moveCamera(CameraUpdateFactory.newLatLngZoom(melbourneLatLng, 3f))
+
+            // Add a listener for polyline clicks that changes the clicked polyline's color.
+            setOnPolylineClickListener { polyline ->
+                // Flip the values of the red, green and blue components of the polyline's color.
+                polyline.color = polyline.color xor 0x00ffffff
+            }
+        }
+
+        // A simple polyline across Australia. This polyline will be mutable.
+        val color = Color.HSVToColor(
```

Comment:
Move this val down into the apply call.

Reply:
Moved down and inlined to the call to color()